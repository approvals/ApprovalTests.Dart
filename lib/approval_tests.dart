import 'dart:convert';
import 'dart:io';
import 'dart:math';
import 'dart:mirrors';

import 'package:dcli/dcli.dart';
import 'package:talker/talker.dart';

part 'src/approvals.dart';
part 'src/writers/approval_text_writer.dart';
part 'src/core/logger/logger.dart';
part 'src/core/utils/utils.dart';
part 'src/core/options.dart';
part 'src/namer/namer.dart';
part 'src/comparator/comparator.dart';
part 'src/core/enums/comporator_ide.dart';
part 'src/comparator/command_line_comparator.dart';
part 'src/comparator/ide_comparator.dart';
part 'src/reporter/doesnt_match_exception.dart';
part 'src/core/utils/extensions.dart';
part 'src/core/utils/converter.dart';
part 'src/core/utils/executable_query.dart';
part 'src/namer/approval_number.dart';
