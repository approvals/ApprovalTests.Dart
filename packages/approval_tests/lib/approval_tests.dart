// ignore_for_file: implementation_imports

import 'dart:convert';
import 'dart:io';
import 'dart:math';

import 'package:diff_match_patch2/diff_match_patch.dart';
import 'package:talker/talker.dart';
import 'package:test_api/src/backend/invoker.dart' show Invoker;

part 'src/approvals.dart';
part 'src/comparator/file_comparator.dart';
part 'src/core/approval_number.dart';
part 'src/core/approval_writer.dart';
part 'src/core/comparator.dart';
part 'src/core/enums/comporator_ide.dart';
part 'src/core/extensions/approval_string_extensions.dart';
part 'src/core/utils/file_path_extractor.dart';
part 'src/core/logger/logger.dart';
part 'src/core/options.dart';
part 'src/core/platform_wrapper.dart';
part 'src/core/reporter.dart';
part 'src/core/scrubber.dart';
part 'src/core/stack_trace_fetcher/fetcher.dart';
part 'src/core/stack_trace_fetcher/stack_trace_fetcher.dart';
part 'src/core/utils/converter.dart';
part 'src/core/utils/executable_query.dart';
part 'src/core/utils/utils.dart';
part 'src/exceptions/doesnt_match_exception.dart';
part 'src/exceptions/file_not_found_exception.dart';
part 'src/exceptions/no_diff_tool_exception.dart';
part 'src/namer/file_namer_options.dart';
part 'src/namer/namer.dart';
part 'src/reporters/command_line/command_line_reporter.dart';
part 'src/reporters/diff_tool/diff_info.dart';
part 'src/reporters/diff_tool/diff_tool_reporter.dart';
part 'src/reporters/diff_tool/diff_tools.dart';
part 'src/scrubbers/date_scrubber.dart';
part 'src/scrubbers/nothing_scrubber.dart';
part 'src/scrubbers/reg_exp_scrubber.dart';
part 'src/writers/approval_text_writer.dart';
part 'src/reporters/git.dart';
