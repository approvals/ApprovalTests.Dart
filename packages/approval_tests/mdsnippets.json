{
  "$schema": "https://raw.githubusercontent.com/SimonCropp/MarkdownSnippets/master/schema.json",
  "Convention": "InPlaceOverwrite",
  "TocLevel": 5
}
