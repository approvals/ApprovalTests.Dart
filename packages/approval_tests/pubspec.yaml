#  * 1.0.0 for a first major release.
#  * 0.1.0 for a breaking release.
#  * 0.0.4 for a minor release.
name: approval_tests
description: Approval Tests implementation in Dart. Inspired by ApprovalTests.
version: 1.0.0
repository: https://github.com/approvals/ApprovalTests.Dart
homepage: https://github.com/approvals/ApprovalTests.Dart.git
issue_tracker: https://github.com/approvals/ApprovalTests.Dart/issues

topics:
  - test
  - approval
  - approvals
  - tests
  - approval-tests

environment:
    sdk: '>=3.0.0 <4.0.0'

dependencies:
  test: ^1.25.2
  talker: ^4.3.2
  diff_match_patch2: ^0.5.0
  test_api: ^0.7.0

dev_dependencies:
  sizzle_lints: ^2.0.2