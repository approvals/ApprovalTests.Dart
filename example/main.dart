import 'package:approval_tests/approval_tests.dart';
import 'package:test/test.dart';

void main() {
  group('Fizz Buzz', () {
    test("Verify all combinations", () {
      Approvals.verifyAll(
        [3, 5, 15],
        options: const Options(
          comparator: IDEComparator(
            ide: ComparatorIDE.visualStudioCode,
          ),
          deleteReceivedFile: true,
        ),
        processor: (items) => fizzBuzz(items).toString(),
      );
    });
  });
}

/// Fizz Buzz kata implementation
List<String> fizzBuzz(int n) {
  final List<String> result = [];

  for (int i = 1; i <= n; i++) {
    // If the current number (`i`) is divisible by both 3 and 5 (or equivalently 15),
    // add the string "FizzBuzz" to the `result` list.
    if (i % 15 == 0) {
      result.add("FizzBuzz");

      // Else, if the current number (`i`) is only divisible by 3,
      // add the string "Fizz" to the `result` list.
    } else if (i % 3 == 0) {
      result.add("Fizz");

      // Else, if the current number (`i`) is only divisible by 5,
      // add the string "Buzz" to the `result` list.
    } else if (i % 5 == 0) {
      result.add("Buzz");

      // If the current number (`i`) is neither divisible by 3 nor 5,
      // add the string representation of that number to the `result` list.
    } else {
      result.add(i.toString());
    }
  }

  return result;
}
