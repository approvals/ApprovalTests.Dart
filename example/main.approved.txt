[[1, 2, Fizz], [1, 2, Fizz, 4, Buzz], [1, 2, Fizz, 4, Buzz, Fizz, 7, 8, Fizz, Buzz, 11, Fizz, 13, 14, FizzBuzz]]